ntc_rosetta/parsers/openconfig/ios/openconfig_system/system.py

```diff
@@ -0,0 +1,309 @@
+            for k, v in dns.items():
+                if k == "#text":
+                    continue
+                yield k, v
+
+    def address(self) -> str:
+        return str(self.yy.key)
+
+    config = DnsServerConfig
+
+
+class DnsServers(Parser):
+    server = DnsServer
+
+    class Yangify(ParserData):
+        path = "/openconfig-system:system/dns/servers"
+
+
+class Dns(Parser):
+    config = DnsConfig
+    servers = DnsServers
+
+    class Yangify(ParserData):
+        path = "/openconfig-system:system/dns"
+
+
+class NtpConfig(Parser):
+    class Yangify(ParserData):
+        path = "/openconfig-system:system/ntp/config"
+
+    def enabled(self) -> bool:
+        if jh.query("ntp", self.yy.native):
+            return True
+        return False
+
+    def ntp_source_address(self) -> str:
+        return None
+
+    def enable_ntp_auth(self) -> bool:
+        if jh.query("ntp.authenticate", self.yy.native):
+            return True
+        return False
+
+
+class NtpServerConfig(Parser):
+    class Yangify(ParserData):
+        path = "/openconfig-system:system/ntp/servers/server/config"
+
+    def address(self) -> str:
+        return str(self.yy.key)
+
+
+class NtpServer(Parser):
+    config = NtpServerConfig
+
+    class Yangify(ParserData):
+        path = "/openconfig-system:system/ntp/servers/server"
+
+        def extract_elements(self) -> Iterator[Tuple[str, Dict[str, Any]]]:
+            ntp_servers = jh.query("ntp.server", self.native) or {}
+            for k, v in ntp_servers.items():
+                if k == "#text":
+                    continue
+                yield k, v
+
+    def address(self) -> str:
+        return str(self.yy.key)
+
+
+class NtpServers(Parser):
+    server = NtpServer
+
+    class Yangify(ParserData):
+        path = "/openconfig-system:system/ntp/servers"
+
+
+class Ntp(Parser):
+    config = NtpConfig
+    servers = NtpServers
+
+    class Yangify(ParserData):
+        path = "/openconfig-system:system/ntp"
+
+
+class SshServerConfig(Parser):
+    class Yangify(ParserData):
+        path = "/openconfig-system:system/ssh-server/config"
+
+    def enable(self) -> bool:
+        # NOTE: This is always true
+        return True
+
+    def protocol_version(self) -> str:
+        version = jh.query('ip.ssh.version."#text"', self.yy.native)
+
+        if version and version == "2":
+            return "V2"
+        elif version and version == "1":
+            return "V1"
+        return "V1_V2"
+
+    def timeout(self) -> int:
+        timeout = jh.query('ip.ssh."time-out"."#text"', self.yy.native)
+        if timeout:
+            return timeout
+        return None
+
+
+class SshServer(Parser):
+    config = SshServerConfig
+
+    class Yangify(ParserData):
+        path = "/openconfig-system:system/ssh-server"
+
+
+class TelnetServerConfig(Parser):
+    class Yangify(ParserData):
+        path = "/openconfig-system:system/telnet-server/config"
+
+    def enable(self) -> bool:
```

Comment:
I think you're right on this. It's always running but only available from the vty linesand must be turned off by excluding telnet from the `input / output` commands under the VTY lines.

Reply:
Yeah, this gets into the guts of the VTY lines.  It's not hurting anything, so I suppose we can leave it here as boilerplate for someone more adventurous.  The minutiae of SSH will provide more bang-for-the-buck.